### ISSUE TITLE
Unable to use /helpers/pull.sh to pull database from multidev environment

### ISSUE THREAD

**Issue by ssteigen** [lando/pantheon#13]
**My Setup**

Lando v3.0.26 on macOS Catalina Version 10.15.7

**The issue**

I'm trying to create a custom lando tooling command that will allow devs to run `lando pull-from-develop` to refresh their local environment from a multidev instance we have that is called `develop`.

I got this idea by looking at the docs here: https://docs.lando.dev/config/pantheon.html#customizing-pantheon-tooling

Here's my tooling entry in my `.lando.yml`:
```
tooling:
  pull-from-develop:
    service: appserver
    cmd: /helpers/pull.sh --code=none --database=develop --files=develop
```

In the process, I noticed there is an issue with the `/helpers/pull.sh` script. It seems like the command is unable to pull the database from a multidev instance.

**Steps to reproduce**

I'm using the https://github.com/pantheon-systems/example-drops-8-composer template as the basis for my project.

I created a brand new D8 Pantheon site and created a brand new multidev instance called `develop`.

```bash
lando init --source cwd --recipe pantheon --pantheon-site mysite
lando start
lando ssh
/helpers/pull.sh --code=none --database=develop --files=none
```

I get the following error:

```
www-data@2f3f990a5e99:/app$ /helpers/pull.sh --code=none --database=develop --files=none
Validating you can pull the database from develop...
 ----------------- ------------------------------------------------------
  ID                develop
  Created           2021-05-10 19:48:26
  Domain            develop-mysite.pantheonsite.io
  Locked            false
  Initialized       true
  Connection Mode   git
  PHP Version       7.3
 ----------------- ------------------------------------------------------
Confirmed!
Destroying all current tables in database if needed...
Making sure your site is awake...
 [notice] OK >> develop-mysite.pantheonsite.io responded
Pulling your database... This miiiiight take a minute
/helpers/pull.sh: line 164: mysite.develop: command not found
   0 B 0:00:00 [   0 B/s] [<=>                                                                                                                                     ]
Checking db pull for expected tables...
Database pull failed...
```

**Comment by reynoldsalec** [comment]
@labboy0276 if you could isolate whether this is purely an interface issue or if there are any deeper issues with the pull script limiting its ability to interact with multidev environments, that would be awesome.

**Comment by labboy0276** [comment]
I am seeing this on `switch` and not `pull` @reynoldsalec : 

Here is 'lando switch`:
```
john@tandem:~/lando/landobot-drupal8 (master)$ lando switch
? Choose a Pantheon account john@thinktandem.io
? Switch environment to? testing
Attempting to login via terminus...
 [notice] Logging in via machine token.
Logged in as john@thinktandem.io
Verifying that you have access to landobot-drupal8...
 ------------------ ------------------------------------------------------------------------------------- 
  ID                 ###                                               
  Name               landobot-drupal8                                                                     
  Label              landobot-drupal8                                                                     
  Created            2019-02-08 18:23:28                                                                  
  Framework          drupal8                                                                              
  Region             United States                                                                        
  Organization       ###                                               
  Plan               Sandbox                                                                              
  Max Multidevs      10                                                                                   
  Upstream           ###: git://github.com/pantheon-systems/drops-8.git  
  Holder Type        user                                                                                 
  Holder ID          ###                                           
  Owner              ###                                               
  Is Frozen?         false                                                                                
  Date Last Frozen   2021-08-11 07:25:53                                                                  
 ------------------ ------------------------------------------------------------------------------------- 
Access confirmed!
Certificate will not expire
Cert is good!
Switching to testing...
Validating you can pull code from testing...
 ----------------- ------------------------------------------ 
  ID                testing                                   
  Created           2021-12-06 17:04:18                       
  Domain            testing-landobot-drupal8.pantheonsite.io  
  Locked            false                                     
  Initialized       true                                      
  Connection Mode   git                                       
  PHP Version       7.3                                       
 ----------------- ------------------------------------------ 
Confirmed!
Fetching origin
Branch 'testing' set up to track remote branch 'testing' from 'origin'.
Switched to a new branch 'testing'
From ssh://codeserver.dev.###.drush.in:2222/~/repository
 * branch                testing    -> FETCH_HEAD
Already up to date.
Validating you can pull the database from testing...
 ----------------- ------------------------------------------ 
  ID                testing                                   
  Created           2021-12-06 17:04:18                       
  Domain            testing-landobot-drupal8.pantheonsite.io  
  Locked            false                                     
  Initialized       true                                      
  Connection Mode   git                                       
  PHP Version       7.3                                       
 ----------------- ------------------------------------------ 
Confirmed!
Destroying all current tables in database if needed... 
Making sure your site is awake...
 [notice] OK >> testing-landobot-drupal8.pantheonsite.io responded
Pulling your database... This miiiiight take a minute
/helpers/pull.sh: line 160: landobot-drupal8.testing: command not found
0.00 B 0:00:00 [0.00 B/s] [<=>                                                 ]
Checking db pull for expected tables...
Database pull failed... 
```

Here is `lando pull`:

```
john@tandem:~/lando/landobot-drupal8 (master)$ lando pull --code=none --database=testing --files=none
? Choose a Pantheon account john@thinktandem.io
Attempting to login via terminus...
 [notice] Logging in via machine token.
Logged in as john@thinktandem.io
Verifying that you have access to landobot-drupal8...
 ------------------ ------------------------------------------------------------------------------------- 
  ID                 ###                                                
  Name               landobot-drupal8                                                                     
  Label              landobot-drupal8                                                                     
  Created            2019-02-08 18:23:28                                                                  
  Framework          drupal8                                                                              
  Region             United States                                                                        
  Organization       ###                                                 
  Plan               Sandbox                                                                              
  Max Multidevs      10                                                                                   
  Upstream           ###: git://github.com/pantheon-systems/drops-8.git  
  Holder Type        user                                                                                 
  Holder ID          ###                                             
  Owner              ###                                             
  Is Frozen?         false                                                                                
  Date Last Frozen   2021-08-11 07:25:53                                                                  
 ------------------ ------------------------------------------------------------------------------------- 
Access confirmed!
Certificate will not expire
Cert is good!
Validating you can pull the database from testing...
 ----------------- ------------------------------------------ 
  ID                testing                                   
  Created           2021-12-06 17:04:18                       
  Domain            testing-landobot-drupal8.pantheonsite.io  
  Locked            false                                     
  Initialized       true                                      
  Connection Mode   git                                       
  PHP Version       7.3                                       
 ----------------- ------------------------------------------ 
Confirmed!
Destroying all current tables in database if needed... 
Making sure your site is awake...
 [notice] OK >> testing-landobot-drupal8.pantheonsite.io responded
Pulling your database... This miiiiight take a minute
 [warning] This environment is in read-only Git mode. If you want to make changes to the codebase of this site (e.g. updating modules or plugins), you will need to toggle into read/write SFTP mode first.
 [notice] Command: landobot-drupal8.testing -- drush sql-dump [Exit: 0]        ]
1.57MiB 0:00:09 [ 171KiB/s] [   <=>                                            ]
Checking db pull for expected tables...
shortcut_set_users
users
users_data
users_field_data
Pull completed successfully!
```

Let me do a little digging on why switch may be failing

**Comment by labboy0276** [comment]
OK, so the issue seems to be that the env vars `LANDO_DB_PULL_COMMAND`, `LANDO_DB_PULL_COMMAND_OPTIONS`, and `LANDO_DB_USER_TABLE` are not set in the pull.sh bash files.  It seems there was a work around for this with a `$FALLBACK_PULL_DB` being set and called via a double pipe `||` command on failure to the line that @abbydrury started to point out.

The issue is that bash is not detecting that the first command has failed with this method.  I tweaked the failure detection as noted in https://github.com/lando/pantheon/pull/83 @reynoldsalec

**Comment by labboy0276** [comment]
After a little more digging, we found the root cause and merged that in.  Fix is in 0.5.2 of this plugin and will be rolled out in the next lando release itself by end of week.
